lib/logstash/outputs/sns.rb

```diff
@@ -70,54 +63,45 @@ def register
   def receive(event)
     return unless output?(event)
 
-    arn     = Array(event["sns"]).first || @arn
-
-    raise "An SNS ARN required." unless arn
-
-    message = Array(event["sns_message"]).first
-    subject = Array(event["sns_subject"]).first || event["host"]
-
-    # Ensure message doesn't exceed the maximum size.
-    if message
-      # TODO: Utilize `byteslice` in JRuby 1.7: http://jira.codehaus.org/browse/JRUBY-5547
-      message = message.slice(0, MAX_MESSAGE_SIZE_IN_BYTES)
+    if (sns_msg = Array(event["sns_message"]).first)
+      send_sns_message(event_arn(event), event_subject(event), sns_msg)
```

Comment:
Actually, my last comment makes no sense for the message field. For that we should pass it through the codec, and whatever comes out goes in. For the subject however, we should either fail if the message is not a string, or save the subject as JSON. 

I'm leaning toward the JSON option.